### PR TITLE
Combat Patrol: War crimes edition

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Adds a couple new weapons and associated loadouts to SOM.

**V-62 Incinerator:**
Goes fwoosh. Shoots further than marine flamers but with a narrower cone, and no watermode.
The loadout has 'Hades' fireproof module, and and burstfire configured V-11.

Shout out to Iamtheguy for the lovely sprite.

https://user-images.githubusercontent.com/7869430/192170191-1fe67e0b-18c1-47eb-b354-66e485bd726f.mp4

**V-71 RPG:**
Rocket propelled grenades, in 6 different flavours
High Explosive
Light Explosive
Thermobaric
HEAT
Incendiary
Radioactive

Shout out to Crow for great sprites, and even doing the mob sprites for me.

https://user-images.githubusercontent.com/7869430/192169698-1af9102f-72bb-4c32-b14f-584c1ed2e4d9.mp4

In combat patrol you'll only get access to incend and rad rockets.
Incend does a WP style fireball.
Rad causes a small (grenade tier) explosion but heavily irradiates anything in the area; the closer you are, the worse the effects.
The initial blast causes a variety of negative effects, and inflicts an irradiated status effect.

The RPG loadout also has a couple of rad grenades, for similar irradiation but no explosion.

Cheers to Zaro for the nade sprite.

**Other stuff:**
Both the flamer and RPG have mob sprites that actually show whether they are loaded as well. I have added support for a binary flag here instead of the existing ammo level code, where you just want to see loaded/unloaded instead of 50% ammo left etc.
Also fixes a long standing bug where weapons that count ammo by shots remaining (i.e. energy weapons) always stated they had full ammo when you examined them.
Fixes a bug where mag fed weapons would have a max_rounds value one higher than it actually was, which had a few knock on effects.
Updated the SOM pistol belt to actually work, in standard and fancy varieties.

## Changelog
:cl:
add: Added the V-62 incinerator for the SOM
add: Added the V-71 RPG for the SOM
add: Added Radiation effects for use by the SOM. War crimes have never been so easy!
fix: Fixed a couple of minor gun ammo bugs
/:cl:

### CODE REVIEW

File: code/modules/projectiles/magazines/specialist.dm [hunk 1]
```diff
@@ -252,6 +252,52 @@
...
+
+/obj/item/ammo_magazine/rocket/som/thermobaric
+	name = "\improper 84mm thermobaric RPG"
+	desc = "A thermobaric warhead for the V-71 rocket launcher. Causes a powerful fuel air explosion over a moderate area."
+	icon_state = "rpg_rocket_thermobaric"
+	default_ammo = /datum/ammo/rocket/som/thermobaric
+	bonus_overlay = "rpg_rocket_thermobaric"
+
+/obj/item/ammo_magazine/rocket/som/rad
+	name = "\improper 84mm  irrad RPG"
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/specialist.dm:287]:
```suggestion
	name = "\improper 84mm tactical nuclear RPG"
```
"Radiological" works too, but it is up to you

File: code/modules/projectiles/magazines/specialist.dm [hunk 2]
```diff
@@ -252,6 +252,52 @@
...
+	desc = "A irrad warhead for the V-71 rocket launcher. Releases a devastating milisecond burst of radiation, debilitating anything caught in the blast radius."
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/specialist.dm:288]:
```suggestion
	desc = "A tactical nuclear warhead for the V-71 rocket launcher. Releases a devastating milisecond burst of radiation, debilitating anything caught in the blast radius."
```

File: code/modules/projectiles/magazines/specialist.dm (hunk 1 above)
Comment (Lumipharon) [reply code/modules/projectiles/magazines/specialist.dm:287]: Calling it nuclear implies it's actually causing a nuclear reaction, this is more of a dirty bomb, it's just a conventional explosive that's also irradiating people.

**Comment by Al-1ce** [comment]
`desc = "You've been irradiated! The effects of the radiation will continue to harm you until purged from your system."`
`er.adjustToxLoss(3)`
`living_owner.adjust_Losebreath(5)`
`if(prob(15)) living_owner.vomit()`
`var/datum/internal_organ/organ = pick(carbon_owner.internal_organs)if(organ)organ.take_damage(5)`

This is concerning. I think most of the community can agree that having a projective which continuously does damage until it's medically purged is bad. Like, really bad. It's not fair nor balanced for the receiving party. As a regular grunt, you are NEVER going to have access to hypervene unless you rob a medic or go corpse-hunting. Which is never reliable.

>but what about satrapine?

Satrapine is a reagent-- it doesn't stay in your system forever, and it doesn't need to be _**exclusively**_ treated to go away.

>why not rely on regular medics?

That's the thing, really. Medics in CP are inherently unreliable. Even if this wasn't an issue, having to completely double-back and sprint to a corpsman to get treated for radiation while you're **_ACTIVELY_** burning from the inside and feeling your organs melt isn't going to be fun.

I don't really think you can switch this in any way that'll make it fair or balanced. Types of damage that can't be cured by normal means or medicine marines would typically have is _incredibly_ rare for a reason.

**Comment by Lumipharon** [comment]
> This is concerning. I think most of the community can agree that having a projective which continuously does damage until it's medically purged is bad. Like, really bad. It's not fair nor balanced for the receiving party. As a regular grunt, you are NEVER going to have access to hypervene unless you rob a medic or go corpse-hunting. Which is never reliable.

Cope and/or seethe. You cannot purge radiation at all, you get irradiated, you ride it out.

> I don't really think you can switch this in any way that'll make it fair or balanced. Types of damage that can't be cured by normal means or medicine marines would typically have is _incredibly_ rare for a reason.

You have 5 minute respawns in combat patrol, and organ damage was horrifically nerfed into a minor inconvenience, organ and clone loss isn't a big deal. Not that being irradiated actually causes enough organ damage to really make any notable effect in most cases.

**Comment by Al-1ce** [comment]
> I would much rather be hit by a rad rocket and live with some long term effects than be hit by an RR shell and almost certainly get killed.

The radiation will kill you either way. I don't see how this is any different from regular RR shells.

**Comment by Lumipharon** [comment]
> > I would much rather be hit by a rad rocket and live with some long term effects than be hit by an RR shell and almost certainly get killed.
> 
> The radiation will kill you either way. I don't see how this is any different from regular RR shells.

It doesn't, you're severely overestimating what it does. It's a max of 3 damage every 2 seconds, and you've got dylo for tox damage, and its for not a terribly long time either.

**Comment by Autisem** [comment]
>V-62 Incinerator:
NO WAY NO FUCKING WAY
afbeelding

**Review by Kuro020 — APPROVED** [review]

**Comment by bowlofcereal** [comment]
i am losing my fucking mind over all this cool shit being added to combat patrol but none of the fuckers in game will vote for it

**Comment by Lumipharon** [comment]
> i am losing my fucking mind over all this cool shit being added to combat patrol but none of the fuckers in game will vote for it

Democracy was a mistake. Lucky you have admin powers to fix mistakes tho

File: code/datums/looping_sounds/machinery_sounds.dm [hunk 3]
```diff
@@ -23,3 +23,19 @@
...
+		list('sound/effects/geiger/low1.ogg'=1, 'sound/effects/geiger/low2.ogg'=1, 'sound/effects/geiger/low3.ogg'=1, 'sound/effects/geiger/low4.ogg'=1),
+		list('sound/effects/geiger/med1.ogg'=1, 'sound/effects/geiger/med2.ogg'=1, 'sound/effects/geiger/med3.ogg'=1, 'sound/effects/geiger/med4.ogg'=1),
+		list('sound/effects/geiger/high1.ogg'=1, 'sound/effects/geiger/high2.ogg'=1, 'sound/effects/geiger/high3.ogg'=1, 'sound/effects/geiger/high4.ogg'=1),
+		list('sound/effects/geiger/ext1.ogg'=1, 'sound/effects/geiger/ext2.ogg'=1, 'sound/effects/geiger/ext3.ogg'=1, 'sound/effects/geiger/ext4.ogg'=1)
+		)
+	mid_length = 2
+	volume = 20
+	max_loops = 5
+	direct = TRUE
+	var/severity = 1
```
Comment (TiviPlus) [inline code/datums/looping_sounds/machinery_sounds.dm:38]: autodoc

File: code/datums/status_effects/debuffs.dm [hunk 4]
```diff
@@ -284,3 +284,44 @@
 /datum/status_effect/spacefreeze/tick()
 	owner.adjustFireLoss(40)
 
+///irradiated mob
+/datum/status_effect/irradiated
+	id = "irradiated"
+	status_type = STATUS_EFFECT_REFRESH
+	tick_interval = 20
+	alert_type = /obj/screen/alert/status_effect/irradiated
+	var/mob/living/carbon/carbon_owner
```
Comment (TiviPlus) [inline code/datums/status_effects/debuffs.dm:293]: Autodoc

File: code/datums/status_effects/debuffs.dm [hunk 5]
```diff
@@ -284,3 +284,44 @@
...
+	//Roulette of bad things
+	if(prob(15))
+		living_owner.adjustCloneLoss(2)
+		to_chat(living_owner, span_warning("You feel like you're burning from the inside!"))
+	else
+		living_owner.adjustToxLoss(3)
+	if(prob(15))
+		living_owner.adjust_Losebreath(5)
+	if(prob(15))
+		living_owner.vomit()
```
Comment (TiviPlus) [inline code/datums/status_effects/debuffs.dm:318]: Lot of RNG but ok

File: code/game/objects/items/explosives/grenades/grenade.dm [hunk 6]
```diff
@@ -92,3 +92,54 @@
...
+	arm_sound = 'sound/weapons/armbomb.ogg' //placeholder
+	hud_state = "grenade_he" //placeholder
+	///The range for the grenade's full effect
+	var/inner_range = 4
+	///The range range for the grenade's weak effect
+	var/outer_range = 7
+	///The potency of the grenade
+	var/rad_strength = 20
+	///geiger counter sound loop
+	var/datum/looping_sound/geiger/geiger_counter
```
Comment (TiviPlus) [inline code/game/objects/items/explosives/grenades/grenade.dm:115]: I dont see the sound getting nulled or deleted

File: code/game/objects/items/storage/holsters.dm [hunk 7]
```diff
@@ -146,6 +146,32 @@
...
+	desc = "This backpack can hold 4 RPGs, in addition to a RPG launcher."
+	icon_state = "som_rocket"
+	item_state = "som_rocket"
+	base_icon = "som_rocket"
+	holsterable_allowed = list(
+		/obj/item/weapon/gun/launcher/rocket/som,
+		/obj/item/weapon/gun/launcher/rocket/som/rad,
+	)
+	bypass_w_limit = list(/obj/item/weapon/gun/launcher/rocket/som,)
+	storage_type_limits = list(/obj/item/weapon/gun/launcher/rocket/som = 1,)
```
Comment (TiviPlus) [inline code/game/objects/items/storage/holsters.dm:160]:
```suggestion
	bypass_w_limit = list(/obj/item/weapon/gun/launcher/rocket/som)
	storage_type_limits = list(/obj/item/weapon/gun/launcher/rocket/som = 1)
```

File: code/modules/mob/living/status_procs.dm [hunk 8]
```diff
@@ -745,3 +745,47 @@
 	else if(amount > 0)
 		M = apply_status_effect(STATUS_EFFECT_MUTED, amount)
 	return M
+
+///////////////////////////////// Rad new //////////////////////////////////
+/mob/living/proc/is_irradiated() //If we're irradiated
+	return has_status_effect(STATUS_EFFECT_IRRADIATED)
```
Comment (TiviPlus) [inline code/modules/mob/living/status_procs.dm:751]: Autodocs for this file

File: code/modules/projectiles/ammo_datums.dm [hunk 9]
```diff
@@ -1764,6 +1770,94 @@ datum/ammo/bullet/revolver/tp44
...
+	///Base strength of the rad effects
+	var/rad_strength = 25
+	///Range for the maximum rad effects
+	var/inner_range = 3
+	///Range for the moderate rad effects
+	var/mid_range = 5
+	///Range for the minimal rad effects
+	var/outer_range = 8
+	///geiger counter sound loop
+	var/datum/looping_sound/geiger/geiger_counter
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1830]: Dont see this getting anulled

**Comment by MereWalker** [comment]
Oh yea it's all coming together just as I envisioned it. Can't wait to absolutely clown on people with the flamer. I just hope they go over cades when fired.

File: code/modules/mob/living/status_procs.dm (hunk 8 above)
Comment (Lumipharon) [reply code/modules/mob/living/status_procs.dm]: done

File: code/datums/status_effects/debuffs.dm (hunk 4 above)
Comment (Lumipharon) [reply code/datums/status_effects/debuffs.dm:294]: done

File: code/datums/looping_sounds/machinery_sounds.dm (hunk 3 above)
Comment (Lumipharon) [reply code/datums/looping_sounds/machinery_sounds.dm:39]: done

**Comment by Lumipharon** [comment]
> Oh yea it's all coming together just as I envisioned it. Can't wait to absolutely clown on people with the flamer. I just hope they go over cades when fired.

Yeah, I refactored flamers so they now go over any obstacle that a projectile could. A lot of weird stuff used to block fire.

File: code/modules/projectiles/ammo_datums.dm (hunk 9 above)
Comment (Lumipharon) [reply code/modules/projectiles/ammo_datums.dm]: Moved the var

File: code/game/objects/items/explosives/grenades/grenade.dm (hunk 6 above)
Comment (Lumipharon) [reply code/game/objects/items/explosives/grenades/grenade.dm]: Moved the var

**Comment by zzurge** [comment]
i cannot nerd out enough about how cool the SOM lineup looks,aesthetically the touch ups have been outstanding.the only shame is how much people cope about the game mode,no one ever votes so we rarely get to use the cool SOM gear

File: code/game/objects/items/explosives/grenades/grenade.dm [hunk 10]
```diff
@@ -92,3 +92,53 @@
...
+	var/rad_strength = 20
+
+/obj/item/explosive/grenade/rad/prime()
+	var/turf/impact_turf = get_turf(src)
+	playsound(impact_turf, 'sound/effects/portal_opening.ogg', 50, 1)
+
+	for(var/mob/living/victim in hearers(outer_range, src))
+		var/strength
+		var/datum/looping_sound/geiger/geiger_counter
+		geiger_counter = new(null, FALSE)
```
Comment (TiviPlus) [inline code/game/objects/items/explosives/grenades/grenade.dm:122]:
```suggestion
		var/datum/looping_sound/geiger/geiger_counter = new(null, FALSE)
```

File: code/modules/mob/living/status_procs.dm [hunk 11]
```diff
@@ -745,3 +745,53 @@
...
+	var/datum/status_effect/irradiated/P = is_irradiated(FALSE)
+	if(amount <= 0)
+		if(P)
+			qdel(P)
+	else
+		if(P)
+			P.duration = world.time + amount
+		else
+			P = apply_status_effect(STATUS_EFFECT_IRRADIATED, amount)
+	return P
```
Comment (TiviPlus) [inline code/modules/mob/living/status_procs.dm:786]: Why P??
PRadiated??
Proper var names pls

File: code/modules/mob/living/status_procs.dm [hunk 12]
```diff
@@ -745,3 +745,53 @@
...
+
+///Modifies irradiation duration
+/mob/living/proc/adjust_radiation(amount, ignore_canstun = FALSE)
+	if(status_flags & GODMODE)
+		return
+	var/datum/status_effect/irradiated/P = is_irradiated(FALSE)
+	if(P)
+		P.duration += amount
```
Comment (TiviPlus) [inline code/modules/mob/living/status_procs.dm:794]: ditto

File: code/modules/projectiles/ammo_datums.dm [hunk 13]
```diff
@@ -1764,6 +1770,93 @@ datum/ammo/bullet/revolver/tp44
...
+	var/mid_range = 5
+	///Range for the minimal rad effects
+	var/outer_range = 8
+
+/datum/ammo/rocket/som/rad/drop_nade(turf/T)
+	playsound(T, 'sound/effects/portal_opening.ogg', 50, 1)
+	for(var/mob/living/victim in hearers(outer_range, T))
+		var/strength
+		var/datum/looping_sound/geiger/geiger_counter
+		geiger_counter = new(null, FALSE)
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1835]:
```suggestion
		var/datum/looping_sound/geiger/geiger_counter = new(null, FALSE)
```

File: code/modules/projectiles/ammo_datums.dm [hunk 14]
```diff
@@ -1764,6 +1770,93 @@ datum/ammo/bullet/revolver/tp44
...
+			strength = rad_strength * 0.7
+			geiger_counter.severity = 3
+		else
+			strength = rad_strength * 0.3
+			geiger_counter.severity = 2
+		irradiate(victim, strength)
+		geiger_counter.start(victim)
+	explosion(T, 0, 0, 3, 0)
+
+/datum/ammo/rocket/som/rad/proc/irradiate(mob/living/victim, strength)
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1849]: doc?

File: code/modules/projectiles/gun_system.dm [hunk 15]
```diff
@@ -499,28 +499,33 @@
 	. += overlay
 
 /obj/item/weapon/gun/update_item_state()
-	if(!CHECK_BITFIELD(flags_gun_features, GUN_SHOWS_AMMO_REMAINING))
+	var/current_state = item_state
+	//If the gun has item states that show how much ammo is remaining
+	if(CHECK_BITFIELD(flags_gun_features, GUN_SHOWS_AMMO_REMAINING))
```
Comment (TiviPlus) [inline code/modules/projectiles/gun_system.dm:504]: use & instead of defines pls

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/ammo_datums.dm (hunk 14 above)
Comment (Lumipharon) [reply code/modules/projectiles/ammo_datums.dm]: whoops, done.

File: code/modules/mob/living/status_procs.dm (hunk 12 above)
Comment (Lumipharon) [reply code/modules/mob/living/status_procs.dm]: done

File: code/modules/projectiles/gun_system.dm (hunk 15 above)
Comment (Lumipharon) [reply code/modules/projectiles/gun_system.dm]: Updated. When if ever is it appropriate to use CHECK_BITFIELD?

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**